pkg/builder/registration_builder.go

```diff
@@ -123,62 +119,82 @@ func NewProbeInfoBuilder(probeType, probeCat string,
+		idField)
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithSupplyChain(supplyChainSet []*proto.TemplateDTO) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithSupplyChain(supplyChainSet []*proto.TemplateDTO,
+) *ProbeInfoBuilder {
 	builder.probeInfo.SupplyChainDefinitionSet = supplyChainSet
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithAccountDefinition(acctDefSet []*proto.AccountDefEntry) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithAccountDefinition(acctDefSet []*proto.AccountDefEntry,
+) *ProbeInfoBuilder {
 	builder.probeInfo.AccountDefinition = acctDefSet
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithFullDiscoveryInterval(fullDiscoveryInSecs int32) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithFullDiscoveryInterval(fullDiscoveryInSecs int32,
+) *ProbeInfoBuilder {
+	// Ignore if the interval is less than DEFAULT_MIN_DISCOVERY_IN_SECS
+	if fullDiscoveryInSecs < pkg.DEFAULT_MIN_DISCOVERY_IN_SECS {
+		return builder
+	}
 	builder.probeInfo.FullRediscoveryIntervalSeconds = &fullDiscoveryInSecs
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithIncrementalDiscoveryInterval(incrementalDiscoveryInSecs int32) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithIncrementalDiscoveryInterval(incrementalDiscoveryInSecs int32,
+) *ProbeInfoBuilder {
+	// Ignore if the interval implies the DISCOVERY_NOT_SUPPORTED value
+	if incrementalDiscoveryInSecs <= pkg.DISCOVERY_NOT_SUPPORTED {
+		return builder
+	}
 	builder.probeInfo.IncrementalRediscoveryIntervalSeconds = &incrementalDiscoveryInSecs
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithPerformanceDiscoveryInterval(performanceDiscoveryInSecs int32) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithPerformanceDiscoveryInterval(performanceDiscoveryInSecs int32,
+) *ProbeInfoBuilder {
+	// Ignore if the interval implies the DISCOVERY_NOT_SUPPORTED value
+	if performanceDiscoveryInSecs <= pkg.DISCOVERY_NOT_SUPPORTED {
+		return builder
+	}
 	builder.probeInfo.PerformanceRediscoveryIntervalSeconds = &performanceDiscoveryInSecs
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithActionPolicySet(actionPolicySet []*proto.ActionPolicyDTO) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithActionPolicySet(actionPolicySet []*proto.ActionPolicyDTO,
+) *ProbeInfoBuilder {
 	builder.probeInfo.ActionPolicy = actionPolicySet
 	return builder
 }
 
-func (builder *ProbeInfoBuilder) WithEntityMetadata(entityMetadataSet []*proto.EntityIdentityMetadata) *ProbeInfoBuilder {
+func (builder *ProbeInfoBuilder) WithEntityMetadata(entityMetadataSet []*proto.EntityIdentityMetadata,
+) *ProbeInfoBuilder {
 	builder.probeInfo.EntityMetadata = entityMetadataSet
 	return builder
 }
 
 func (builder *ProbeInfoBuilder) Create() *proto.ProbeInfo {
+	// Assert that the full discovery interval is set
+	if *builder.probeInfo.FullRediscoveryIntervalSeconds <= 0 {
```

Comment:
I think it is better to log a warning msg, and tells that the final value of it.